edx_prefectutils/hubspot_leads.py

```diff
@@ -1,68 +1,153 @@
 # writing, our longest leads list is 14 items long (and only 4 users have more than 10). I think we're OK for now.
 # https://www.braze.com/docs/api/objects_filters/user_attributes_object/
-MASTERS_LEADS_LIMIT = 25
+ARRAY_LIMIT = 25
+
+# https://www.braze.com/docs/api/endpoints/user_data/post_user_track/
+ATTRIBUTES_LIMIT = 75
+
+
+def _make_custom_user_attrs(row):
+    """Dict of attributes that are owned by this script"""
+    # Masters are separated by semicolons in the snowflake database
+    masters_leads = row.masters_leads.split(';') if row.masters_leads else []
+    if len(masters_leads) > ARRAY_LIMIT:
+        raise ValueError(
+            f'At most {ARRAY_LIMIT} masters leads at a time can be set '
+            f'in Braze, but we have {len(masters_leads)} for hubspot user {row.hs_object_id}'
+        )
+
+    # Programs are separated by semicolons in the snowflake database
+    braze_programs = row.braze_programs.split(';') if row.braze_programs else []
+    if len(braze_programs) > ARRAY_LIMIT:
+        raise ValueError(
+            f'At most {ARRAY_LIMIT} program interests at a time can be set '
+            f'in Braze, but we have {len(braze_programs)} for hubspot user {row.hs_object_id}'
+        )
+
+    # Yes these are inconsistent. But it's a matter of historical field names.
+    return {
+        'HubSpot Audience': row.audience,
+        'HubSpot Contact': row.braze_hs_contact,
+        'HubSpot Contact ID': row.hs_object_id,
+        'HubSpot Create Date': row.createdate.isoformat(),
+        'HubSpot - First Conversion': row.first_conversion_event_name,
+        'HubSpot - Lead Source': row.lead_source_deal_value,
+        'HubSpot - Lead Source Detail': row.lead_source,  # yes, the bare `lead_source` is actually the detail
+        'HubSpot Level of Education': row.highest_level_of_education,
+        'HubSpot Masters Interests': masters_leads,
+        'HubSpot Program Interests': braze_programs,
+        'HubSpot - utm_campaign': row.utm_campaign,
+        'HubSpot - utm_medium': row.utm_medium,
+        'HubSpot - utm_source': row.utm_source,
+    }
+
+
+def _make_standard_user_attrs(row):
+    """Dict of attributes that are standard braze ones - that we likely only need to set for unregistered users"""
+    return {
+        'country': row.country_iso_code,
+        'email': row.email,
+        'first_name': row.firstname,
+        'home_city': row.city,
+        'last_name': row.lastname,
+        'phone': row.phone,
+    }
 
 
 @task
 @backoff.on_exception(backoff.expo,
                       ProgrammingError,
                       max_tries=3)
 def sync_hubspot_leads_to_braze(
-    sf_credentials: snowflake.SFCredentials,
+    sf_credentials: SFCredentials,
     sf_role: str,
     sf_database: str,
     sf_schema: str,
     braze_api_key: str,
     braze_api_server: str,
 ):
-    logger = prefect.context.get("logger")
+    logger = prefect.context.get('logger')
+    columns = [
```

Comment:
Does it make sense to move this to a constant ?

Reply:
Hmm... as long as this function is the only one that uses this list, my vote would be to leave it in the function scope rather than bring it out to the module scope. But I can move & capitalize it easy enough, if you think it would be cleaner if it were a module constant.